enumerables.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# This is an enumerable module
+module Enumerables
+  def my_each
+    arr.length.time { |i| yield(arr[i]) }
+    arr
+  end
+
+  def my_each_with_index
+    arr.length.time { |i| yield(arr[i], i) }
+    arr
+  end
```

Comment:
- [ ] It returns an enumerator if no block is given. Just like with the my_each above the same case applies to my each with index.

- [ ]  when a block is given when self is an array returns the array itself after calling the given block once for each element in self.

- [ ] Also make sure it works for all other classes like a hash and a range etc

Kindly refer [refer](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index)